src/ucp/wireup/wireup.c

```diff
@@ -551,20 +556,27 @@ void ucp_wireup_remote_connected(ucp_ep_h ep)
 }
 
 static UCS_F_ALWAYS_INLINE unsigned
-ucp_ep_err_mode_init_flags(ucp_err_handling_mode_t err_mode)
+ucp_wireup_ep_init_flags(const ucp_wireup_msg_t *msg)
 {
-    return (err_mode == UCP_ERR_HANDLING_MODE_PEER) ?
-           UCP_EP_INIT_ERR_MODE_PEER_FAILURE : 0;
+    unsigned ep_init_flags = 0;
+
+    if (msg->flags & UCP_EP_CONFIG_KEY_FLAG_ERR_HANDLING_MODE_PEER) {
+        ep_init_flags |= UCP_EP_INIT_ERR_MODE_PEER_FAILURE;
+    }
+
+    return ep_init_flags;
 }
 
 static UCS_F_NOINLINE void
 ucp_wireup_process_pre_request(ucp_worker_h worker, ucp_ep_h ep,
                                const ucp_wireup_msg_t *msg,
                                const ucp_unpacked_address_t *remote_address)
 {
-    unsigned ep_init_flags = UCP_EP_INIT_CREATE_AM_LANE |
-                             UCP_EP_INIT_CM_WIREUP_CLIENT |
-                             ucp_ep_err_mode_init_flags(msg->err_mode);
+    ucp_err_handling_mode_t err_mode =
+            ucp_ep_config_key_flags_err_handling_mode(msg->flags);
```

Comment:
i think wire compat is broken here. Previously `msg->flags` was compated by value, not it is treated as a flag

Reply:
yes, this is a problem…
but I think we can avoid this change, I’ll check if it is possible